lib/data/locationConstraintParser.js

```
@@ -106,6 +106,9 @@ function parseLC() {
             }
             clients[location] = locationObj.type === 'gcp' ?
                 new GcpClient(clientConfig) : new AwsClient(clientConfig);
+            if (locationObj.type === 'aws_s3') {
```

should this `if` check also verify that the `aws_s3` location is an actual `aws` location before performing the setup operation?

---

That's true. Let me see what we have.

---

I checked the code and if it's non-AWS like DigitalOcean, Wasabi etc, at the best they will return NotImplemented and it will fail silently.

---

For `S3 Ring`, `cloudserver` does have the `getBucketLocation` implemented and will return a location; wouldn't the setup code would overwrite the client with an incorrect endpoint?

---

Ah yeah you are right, since we do implement locations. I could probably check if endpoint === 's3.amazonaws.com' and fix it.